### PR TITLE
JAMES-2961 Document outgoing SMTP security

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RemoteDelivery.java [hunk 1]
```diff
@@ -114,6 +114,15 @@
  * the ability to perform their own problem resolutions.</li>
  * <li><b>debug</b> (optional) - a Boolean (true/false) indicating whether debugging is on. Default is false.</li>
  * </ul>
+ * <br/>
+ * <b>Security:</b><br/>
+ * You can use the <i>mail.smtp.ssl.enable</i> javax property describe above to force SMTP outgoing delivery to default to SSL
```
Comment (rouazana) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RemoteDelivery.java]: we should recommend using tls.enable instead, no?

Comment (chibenwa) [reply server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RemoteDelivery.java]: What do you mean by "TLS"? How do you achieve this using javax.mail.smtp? What precise property would you advice users to use?

Comment (rouazana) [reply server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RemoteDelivery.java]: sorry, I meant using `<startTLS>true</startTLS>` in RemoteDelivery mailet configuration.
(and yes we need to properly document what can be configured by the RemoteDelivery mailet, what can be configured by mail.smtp properties, and how it interracts between each other).

Comment (Arsnael) [reply server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RemoteDelivery.java]: s/describe/described

File: src/site/xdoc/server/feature-security.xml [hunk 2]
```diff
@@ -31,6 +31,9 @@
     
     <p>SMTP Auth and "Verify Identity" options are enabled when you install James (<a href="config-smtp-lmtp.html">read more</a>).</p>
 
+    <p>SMTP outgoing traffic can be transmitted via SSL by default. Check <a href="http://james.apache.org/server/dev-provided-mailets.html##RemoteDelivery">RemoteDelivery</a> documentation for
```
Comment (trantienduchn) [inline src/site/xdoc/server/feature-security.xml]: `http://james.apache.org/server/dev-provided-mailets.html#RemoteDelivery` with only **one** `#`

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RemoteDelivery.java [hunk 3]
```diff
@@ -114,6 +114,15 @@
...
+ * encrypted traffic. <br/>
+ * When enabling SSL, you might need to specify <i>mail.smtp.ssl.checkserveridentity</i> and <i>mail.smtp.ssl.trust</i>
+ * properties. You can also control ciphersuites and protocols via <i>mail.smtp.ssl.ciphersuites</i> and
+ * <i>mail.smtp.ssl.protocols</i> properties.<br/>
+ * Read <a href="http://java.sun.com/products/javamail/javadocs/com/sun/mail/smtp/package-summary.html"><code>com.sun.mail.smtp</code></a>
```
Comment (trantienduchn) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RemoteDelivery.java]: I checked, this site is not available, consider this one https://javaee.github.io/javamail/docs/api/com/sun/mail/smtp/package-summary.html

**Comment by Arsnael** [comment]
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project apache-james-mailbox-event-rabbitmq: There was a timeout or other error in the fork -> [Help 1]
```
? test this please

**Comment by blackheaven** [comment]
Commit name issue

**Comment by chibenwa** [comment]
> Commit name issue 

I don't understand this cryptic comment.

A bit more context would be appreciated @blackheaven

**Comment by chibenwa** [comment]
> Shouldn't we enable starttls and starttls.required by default?

This require trust configuration. I fear people might get this wrong and start complain. Wildcard accept would be OK to me but this might deserve a mailing list question.

**Comment by chibenwa** [comment]
>  I guess you cannot use ssl and tls at the same time (but it's just a guess, it should be checked)

No idea

**Comment by chibenwa** [comment]
test this please

**Comment by mbaechler** [comment]
> This require trust configuration

Default Java behavior is not enough? I mean, it does certificate validation with know roots CA already, what would we lack?

**Comment by chibenwa** [comment]
> Default Java behavior is not enough? I mean, it does certificate validation with know roots CA already, what would we lack?

I miss knowledge on this I must recognize.

**Comment by mbaechler** [comment]
> > Default Java behavior is not enough? I mean, it does certificate validation with know roots CA already, what would we lack?
> 
> I miss knowledge on this I must recognize.

It should work as expected but writing integration test for it is another thing entirely
It would require to add a CA into ssl configuration

**Comment by Arsnael** [comment]
```
[f43fd0de2dd767c59f6c603aeb51027f85a1435a] [ERROR]   MemorySpamAssassinContractTest.deletingSpamMessageShouldNotImpactSpamAssassinState » ConditionTimeout
[f43fd0de2dd767c59f6c603aeb51027f85a1435a] [ERROR]   MemorySpamAssassinContractTest.movingAMailToTrashShouldNotImpactSpamassassinLearning » ConditionTimeout
[f43fd0de2dd767c59f6c603aeb51027f85a1435a] [ERROR]   MemorySpamAssassinContractTest.spamShouldBeDeliveredInSpamMailboxWhenSameMessageHasAlreadyBeenMovedToSpam » ConditionTimeout
```
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**